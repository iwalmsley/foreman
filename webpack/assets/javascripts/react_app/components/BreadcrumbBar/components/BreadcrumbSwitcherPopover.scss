@import '../../../common/colors';

.popover-content {
  max-height: $switcher-max-height;
  overflow-y: auto;
  max-width: $switcher-max-width;
  min-width: $switcher-max-width;

  .breadcrumb-switcher-popover-error {
    color: #f00;
  }

  .list-group-item {
    padding: 5px 3px;

    &:hover {
      color: #fff;
      background-color: #39a5dc;
      border-color: #39a5dc;
    }
  }

  .no-border {
    border: none;
  }
}

#breadcrumb-switcher-popover {
  z-index: 1000 !important;
}
