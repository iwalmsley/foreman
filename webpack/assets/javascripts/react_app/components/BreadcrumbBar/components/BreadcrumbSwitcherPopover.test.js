import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';

import BreadcrumbSwitcherPopover from './BreadcrumbSwitcherPopover';
import {
  breadcrumbSwitcherLoading,
  breadcrumbSwitcherLoaded,
  breadcrumbSwitcherLoadedWithPagination,
} from '../BreadcrumbBar.fixtures';

const fixtures = {
  'render loading state': { id: 'some-id', ...breadcrumbSwitcherLoading },
  'render resources list': { id: 'some-id', ...breadcrumbSwitcherLoaded },
  'render resources list with pagination': {
    id: 'some-id',
    ...breadcrumbSwitcherLoadedWithPagination,
  },
};

describe('BreadcrumbSwitcherPopover', () =>
  testComponentSnapshotsWithFixtures(BreadcrumbSwitcherPopover, fixtures));
