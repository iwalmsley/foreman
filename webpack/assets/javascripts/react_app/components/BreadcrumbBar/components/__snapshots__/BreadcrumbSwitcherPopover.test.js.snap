// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbSwitcherPopover render loading state 1`] = `
<Popover
  bsClass="popover"
  className="breadcrumb-switcher-popover"
  id="some-id"
  placement="right"
>
  <div
    className="breadcrumb-switcher-popover-loading text-center"
  >
    <Icon
      name="spinner"
      size="lg"
      spin={true}
      type="fa"
    />
  </div>
</Popover>
`;

exports[`BreadcrumbSwitcherPopover render resources list 1`] = `
<Popover
  bsClass="popover"
  className="breadcrumb-switcher-popover"
  id="some-id"
  placement="right"
>
  <React.Fragment>
    <ListGroup
      bsClass="list-group"
      className="scrollable-list"
    >
      <ListGroupItem
        active={false}
        bsClass="list-group-item"
        className="no-border"
        href="#"
        id="1-Host 1"
        key="1-Host 1"
        listItem={false}
        onClick={[Function]}
      >
        <EllipisWithTooltip>
          Host 1
        </EllipisWithTooltip>
      </ListGroupItem>
      <ListGroupItem
        active={false}
        bsClass="list-group-item"
        className="no-border"
        href="#"
        id="2-Host 2"
        key="2-Host 2"
        listItem={false}
        onClick={[Function]}
      >
        <EllipisWithTooltip>
          Host 2
        </EllipisWithTooltip>
      </ListGroupItem>
      <ListGroupItem
        active={false}
        bsClass="list-group-item"
        className="no-border"
        href="#"
        id="3-Host 3 with a very long name"
        key="3-Host 3 with a very long name"
        listItem={false}
        onClick={[Function]}
      >
        <EllipisWithTooltip>
          Host 3 with a very long name
        </EllipisWithTooltip>
      </ListGroupItem>
      <ListGroupItem
        active={false}
        bsClass="list-group-item"
        className="no-border"
        id="4-Host 4"
        key="4-Host 4"
        listItem={false}
        onClick={[Function]}
      >
        <EllipisWithTooltip>
          Host 4
        </EllipisWithTooltip>
      </ListGroupItem>
      <ListGroupItem
        active={false}
        bsClass="list-group-item"
        className="no-border"
        href="#"
        id="5-Host 5"
        key="5-Host 5"
        listItem={false}
        onClick={[Function]}
      >
        <EllipisWithTooltip>
          Host 5
        </EllipisWithTooltip>
      </ListGroupItem>
    </ListGroup>
    <Pager
      baseClassName="content-view-pf-pagination"
      className="pager-sm"
      disableNext={true}
      disablePrevious={true}
      messages={
        Object {
          "nextPage": "Next",
          "previousPage": "Previous",
        }
      }
      onNextPage={[Function]}
      onPreviousPage={[Function]}
    />
  </React.Fragment>
</Popover>
`;

exports[`BreadcrumbSwitcherPopover render resources list with pagination 1`] = `
<Popover
  bsClass="popover"
  className="breadcrumb-switcher-popover"
  id="some-id"
  placement="right"
>
  <React.Fragment>
    <ListGroup
      bsClass="list-group"
      className="scrollable-list"
    >
      <ListGroupItem
        active={false}
        bsClass="list-group-item"
        className="no-border"
        href="#"
        id="1-Host 1"
        key="1-Host 1"
        listItem={false}
        onClick={[Function]}
      >
        <EllipisWithTooltip>
          Host 1
        </EllipisWithTooltip>
      </ListGroupItem>
      <ListGroupItem
        active={false}
        bsClass="list-group-item"
        className="no-border"
        href="#"
        id="2-Host 2"
        key="2-Host 2"
        listItem={false}
        onClick={[Function]}
      >
        <EllipisWithTooltip>
          Host 2
        </EllipisWithTooltip>
      </ListGroupItem>
      <ListGroupItem
        active={false}
        bsClass="list-group-item"
        className="no-border"
        href="#"
        id="3-Host 3 with a very long name"
        key="3-Host 3 with a very long name"
        listItem={false}
        onClick={[Function]}
      >
        <EllipisWithTooltip>
          Host 3 with a very long name
        </EllipisWithTooltip>
      </ListGroupItem>
      <ListGroupItem
        active={false}
        bsClass="list-group-item"
        className="no-border"
        id="4-Host 4"
        key="4-Host 4"
        listItem={false}
        onClick={[Function]}
      >
        <EllipisWithTooltip>
          Host 4
        </EllipisWithTooltip>
      </ListGroupItem>
      <ListGroupItem
        active={false}
        bsClass="list-group-item"
        className="no-border"
        href="#"
        id="5-Host 5"
        key="5-Host 5"
        listItem={false}
        onClick={[Function]}
      >
        <EllipisWithTooltip>
          Host 5
        </EllipisWithTooltip>
      </ListGroupItem>
    </ListGroup>
    <Pager
      baseClassName="content-view-pf-pagination"
      className="pager-sm"
      disableNext={false}
      disablePrevious={false}
      messages={
        Object {
          "nextPage": "Next",
          "previousPage": "Previous",
        }
      }
      onNextPage={[Function]}
      onPreviousPage={[Function]}
    />
  </React.Fragment>
</Popover>
`;
